llvm/lib/SYCLLowerIR/ModuleSplitter.cpp

```diff
@@ -478,13 +484,18 @@ void ModuleSplitterBase::verifyNoCrossModuleDeviceGlobalUsage() {
         continue;
       }
       if (auto *F = dyn_cast<const Function>(U)) {
-        if (EntryPointModules.count(F))
-          CheckEntryPointModule(F);
+        if (EntryPointModules.count(F)) {
```

Comment:
A suggestion.
Going forward it might help if you can split such submissions into multiple commits: (1) Code motion without changes to error handling and (2) Handle changes to error handling. it might make it easy to triage errors later on. Thanks